### ISSUE TITLE
vault_pki_secret_backend_cert renew

### ISSUE THREAD

**Issue by mitchellmaler** [terraform-providers/terraform-provider-vault#353]
We are currently using vault_pki_secret_backend_cert to generate certs either to place on a server or to use locally in terraform for other access. 

The issue I have is there is no way to renew the certificate if it is expired unless you taint the resource. 

Since Vault returns the TTL for the cert and also the certificate contains the expire date their should be a trigger on this date to force a recreate of the cert.

**Comment by cvbarros** [comment]
@mitchellmaler as I understand, do you mean for the resource itself to control the certificate lifecycle if it expires? 
And then on subsequent plan/applies it would detect and automatically renew itself or take some configurable action?

**Comment by estein9825** [comment]
I have the same issue with changing the TTL.  It doesn't generate a new cert, even though Terraform recognizes that the TTL has changed. I see this behavior in all modules that offer a max_ttl and/or ttl attribute.

**Comment by mitchellmaler** [comment]
@cvbarros Yes if the certificate resource is expired or will soon it should renew it and optionally revoke the old one if it was not expired but really close too.

**Comment by joemiller** [comment]
I have proposed a solution over at https://github.com/terraform-providers/terraform-provider-vault/issues/353

**Comment by cpick** [comment]
The `auto_renew` and `min_seconds_remaining` settings solved this problem for me, *however* since `auto_renew` was [not documented](https://www.terraform.io/docs/providers/vault/r/pki_secret_backend_cert.html) I thought it was broken until I learned about it from #386.

Would it be possible to mention that `auto_renew = true` is necessary for `min_seconds_remaining` to have any effect?

**Comment by joemiller** [comment]
@cpick That was my oversight. I'll send a docs PR to add auto_renew

**Comment by joemiller** [comment]
@cpick Thanks for catching that. PR to fix the docs - https://github.com/terraform-providers/terraform-provider-vault/pull/479

**Comment by jbcraig** [comment]
When auto_renew triggers, the **expiration** field isn't marked as a pending change.  This requires me to run the apply twice before all my resources become consistent.

**Comment by amarjothi** [comment]
Please can someone explain how this auto_renew works. I have set it to true when creating. How do we retrieve the cert and upload it to AWS. It is attached to ALB. Will it be a new cert and extended ttl?

**Comment by adkam5** [comment]
@joemiller 
Hi Joe!
Recently I have tried to use this feature with these params:
`resource "vault_pki_secret_backend_cert" "test-cert-auto-update" {
  backend     = "pki"
  name        = "testca"
  ttl         = "2h"
  auto_renew  = true
  min_seconds_remaining = 3600
  common_name = "test-cert-auto-update.testca"
}`

I have used shorter periods to test this (in real case I would do a monthly update)
and I have expected, that the certificate will be automatically replaced on next next terraform apply.
initial deployment: 14:00 CET -> resource with certificates created
deployment 14:45 CET -> resources/certificates not replaced
deployment 15:15: CET -> resources/certificates not replaced (here I would have expected, that resource/certificate will be replaced)
